lib/PuppeteerSharp/ChromeTargetManager.cs

```diff
@@ -125,14 +137,16 @@ private void StoreExistingTargetsForInit()
             }
         }
 
-        private async void OnMessageReceived(object sender, MessageEventArgs e)
+        private void OnMessageReceived(object sender, MessageEventArgs e)
         {
             try
             {
                 switch (e.MessageID)
                 {
                     case "Target.attachedToTarget":
-                        await OnAttachedToTarget(sender, e.MessageData.ToObject<TargetAttachedToTargetResponse>(true)).ConfigureAwait(false);
+#pragma warning disable CS4014
+                        OnAttachedToTarget(sender, e.MessageData.ToObject<TargetAttachedToTargetResponse>(true)).ConfigureAwait(false);
```

Comment:
It doesn't change the handling of exceptions before the first `await` in `OnAttachedToTarget`. After that point, any exceptions would be unhandled and wouldn't be logged, and the connection wouldn't be closed.

I will run some tests forcing exceptions in that code, and improve this handling.

Reply:
If we don't await `OnAttachedToTarget` we must wrap all the code inside `OnAttachedToTarget` in a try-catch and log the error.

Reply:
Did this one slip through?
It would seem more safe to do it like `OnAttachedToTargetHandlingExceptionsAsync` to avoid any exceptions being thrown in the wild.

Reply:
E.g. try running the code below.
Calling `AsyncVoid.A()` I get
![image](https://github.com/hardkoded/puppeteer-sharp/assets/919634/fe85ff4d-4e5f-4e87-ae3e-c66a59748c05)

While when calling `AsyncTask.A()` I get
![image](https://github.com/hardkoded/puppeteer-sharp/assets/919634/79f9cf18-8e68-433c-9cc7-7a3dc5f93de4)


```cs
using System;
using System.Threading.Tasks;

TaskScheduler.UnobservedTaskException += TaskScheduler_UnobservedTaskException;

try
{
    Console.WriteLine("Main try");
    AsyncVoid.A();

    Console.WriteLine("---");
    await Task.Delay(1000);
    Console.WriteLine("---");

    AsyncTask.A();

    Console.WriteLine("---");
    await Task.Delay(1000);
}
catch (Exception e)
{
    Console.WriteLine("Main catch");
    Console.WriteLine(e);
}
finally
{
    Console.WriteLine("Main finally");
}

static void TaskScheduler_UnobservedTaskException(object sender, UnobservedTaskExceptionEventArgs e)
{
    Console.WriteLine("Unobserved exception!");
    Console.WriteLine(e.Exception);
}

static class AsyncVoid
{
    public static void A()
    {
        try
        {
            Console.WriteLine("AsyncVoid sync try");
            B();
        }
        catch (Exception e)
        {
            Console.WriteLine("AsyncVoid sync catch");
            Console.WriteLine(e);
        }
        finally
        {
            Console.WriteLine("AsyncVoid sync finally");
        }
    }

    private static async void B()
    {
        try
        {
            Console.WriteLine("AsyncVoid async try");
            await Task.Delay(100);
            throw new TimeoutException();
        }
        catch
        {
            Console.WriteLine("AsyncVoid async catch");
            throw;
        }
    }
}

static class AsyncTask
{
    public static void A()
    {
        try
        {
            Console.WriteLine("AsyncTask sync try");
            _ = B();
        }
        catch (Exception e)
        {
            Console.WriteLine("AsyncTask sync catch");
            Console.WriteLine(e);
        }
        finally
        {
            Console.WriteLine("AsyncTask sync finally");
        }
    }

    private static async Task B()
    {
        try
        {
            Console.WriteLine("AsyncTask async try");
            await Task.Delay(100);
            throw new TimeoutException();
        }
        catch
        {
            Console.WriteLine("AsyncTask async catch");
            throw;
        }
    }
}
```

Reply:
@jnyrup `OnAttachedToTargetHandlingExceptionsAsync` has a try/catch. It won't throw an exception.

Reply:
Sorry, meant `OnTargetDestroyed`.

Reply:
Yes. We do need a try/catch there @jnyrup :)

Reply:
Already done
https://github.com/hardkoded/puppeteer-sharp/pull/2289/commits/f5ac62fee3c0f6d5aec0227e70d71b2ef9bcc346?w=1